.github/workflows/smoke.yml

```diff
@@ -111,8 +112,12 @@ jobs:
 
         echo "::set-output name=go-version::$(grep '^go ' go.mod | awk '{print $2}')"
         echo "::set-output name=base-juju-version::$(juju version | cut -d '-' -f 1)"
-        echo "::set-output name=upstream-juju-version::$(grep -r "const version =" version/version.go | sed -r 's/^const version = \"(.*)\"$/\1/')"
-        echo "::set-output name=juju-db-version::$(grep -r 'DefaultJujuDBSnapChannel =' controller/config.go | sed -r 's/^\s*DefaultJujuDBSnapChannel = \"([[:digit:]]+\.[[:digit:]]+(\.[[:digit:]]+){0,1})\/.*\"$/\1/')"
+        upstreamJujuVersion=$(grep -r "const version =" version/version.go | sed -r 's/^const version = \"(.*)\"$/\1/')
+        echo "::set-output name=upstream-juju-version::${upstreamJujuVersion}"
+        currentStableChannel="$(echo $upstreamJujuVersion | cut -d'.' -f1,2)/stable"
+        currentStableVersion=$(snap info juju | yq ".channels[\"$currentStableChannel\"]" | cut -d' ' -f1)
```

Comment:
Wait so why do we need `upstreamJujuVersion` **and** `currentStableVersion`?

Reply:
Because we have to upgrade from currentStableVersion then upstreamJujuVersion.
You see the two upgrade steps?